accepted/2022/Multiple-Equivalent-Framework-Support-TFM-As-Aliases.md

```diff
@@ -0,0 +1,397 @@
+Some examples are:
+
+- The .NET SDK, as part of build, as previously discussed.
+- Any `dotnet` CLI command with a `--frameworks` or `-f` argument, such as `dotnet build`, `dotnet publish`, `dotnet test`.
+- `dotnet list package`.
+- Numerous features in Visual Studio.
+  - Test Explorer.
+  - Text editor (left-most dropdown, for TFM specific Intellisense).
+  - Solution Explorer's Dependencies node.
+
+### Lock file changes
+
+NuGet's "repeatable build" feature adds a package lock file to the project directory.
+The restore has a lock file has a similar schema to the assets file, so that schema would also need to be amended.
+Fortunately when Central Package Management and its transitive pinning was introduced we introduced the concept of a PackagesLockFile version successfully.
+We'd just add a version 3.
+
+NuGet would add version 4 of the packages lock file, which will [pivot on the target framework alias, just as assets files will](#target-framework-pivots).
+
+NuGet's intra restore caching is based on the existing of a single framework.
+It is difficult to predict the amount of work that'd be required to implement this correctly, but it'll certainly involve public API changes to NuGet libraries.
+(NuGet ships libraries, NuGet.exe, VS and dotnet.exe)
+
+### Pack changes
+
+For the first version of this feature, projects that have more than one `TargetFramework` (inner-build) resolve to the same "NuGet Framework" will result in a pack error, therefore preventing these projects from being packed.
+See [Extending Pack, under Future Possibilities](#extending-pack) for more information.
+
+### Visual Studio challenges
+
+Currently the PM UI does not have any support for multi targeting.
+This isn't anything new, but it can affect the experience of customers moving to SDK projects for the first time.
+Especially for customers who are not comfortable hand editing XML or MSBuild files.
+
+Another concern are all the APIs for displaying the dependencies, transitive dependencies and Get Installed Packages APIs in NuGet.VisualStudio.Contracts.
+All of these APIs may require an incremental addition.
+
+### Project to Project references
+
+MSBuild has a [`ProjectReference` protocol](https://github.com/dotnet/msbuild/blob/25fdeb3c8c2608f248faec3d6d37733ae144bbbb/documentation/ProjectReference-Protocol.md), which explains how `ProjectReferences` work.
+MSBuild calls the `GetReferenceNearestTargetFrameworkTask`, which NuGet implements, in order to obtain best TFM matching.
+
+Currently `GetReferenceNearestTargetFrameworkTask` is implemented by running NuGet's "nearest match" algorithm.
+This feature would extend it to first try nearest match, and if there is more than one matching `TargetFramework`, then look for a `TargetFramework` that has an exact name match in both projects.
+If there are more than one "nearest" framework, but no exact name match, then for the first version of this feature, NuGet will report an error.
```

Comment:
I'm interpreting this as supporting a case where you have a project that uses aliases referencing a project that's not using alias.

So if I have project 1 with TFs: `net6.0-a;net6.0-b` (where both are aliases for `net6.0`) and it referenced project 2 with TFs: `net6.0;netstandard2.0` project1 could reference project 2 and select `net6.0` because there is a simgle nearest framework.

If that's true this is a great middle ground for enabling aliases in a way that isn't viral.